### ISSUE TITLE
No warning reported returning nullable version of unconstrained type parameter

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#46150]
A warning should be reported for the return value:
```C#
#nullable enable

class Program
{
    static T F<T, U>(U u) where U : T?
    {
        return u; // warning?
    }
}
```
